src/data/repositories/GlassUploadsProgramRepository.ts

```diff
@@ -0,0 +1,669 @@
+                ? [
+                      {
+                          dataElement: uploadsDHIS2Ids.asyncImportSummaries,
+                          value: asyncImportSummariesId,
+                      },
+                  ]
+                : []),
+        ];
+
+        const allDataValues: D2TrackerEventToPost["dataValues"] = [...dataValues, ...importSummariesDataValues];
+
+        return {
+            event: upload.id,
+            program: AMR_GLASS_PROE_UPLOADS_PROGRAM_ID,
+            programStage: AMR_GLASS_PROE_UPLOADS_PROGRAM_STAGE_ID,
+            orgUnit: upload.orgUnit,
+            occurredAt: periodToYearMonthDay(upload.period),
+            dataValues: allDataValues,
+        };
+    }
+
+    private getImportSummary(eventId: Id, dataElementId: Id): FutureData<ImportSummaryErrors> {
+        return apiToFuture(
+            this.api.get<ImportSummaryErrors>(`/tracker/events/${eventId}/dataValues/${dataElementId}/file`)
+        ).flatMap(importSummary => {
+            return Future.success(importSummary);
+        });
+    }
+
+    private getAsyncImportSummaries(eventId: Id, dataElementId: Id): FutureData<ImportSummary[]> {
+        return apiToFuture(
+            this.api.get<ImportSummary[]>(`/tracker/events/${eventId}/dataValues/${dataElementId}/file`)
+        ).flatMap(asyncImportSummaries => {
+            return Future.success(asyncImportSummaries);
+        });
+    }
+
+    private saveEventDataValueFile(payload: UploadsFormData): FutureData<Id> {
+        return apiToFuture(
+            this.api.post<PartialSaveFileResourceResponse>("/fileResources", undefined, payload, {
+                requestBodyType: "raw",
+            })
+        ).flatMap(fileUploadResult => {
+            const fileResourceId = fileUploadResult.response?.fileResource?.id;
+
+            if (!fileResourceId) {
+                return Future.error("Error when saving event data value file");
+            }
+
+            return Future.success(fileResourceId);
+        });
+    }
+
+    private getEventsIdsChunked(ids: Id[], chunkSize: number): FutureData<D2TrackerEvent[]> {
+        const chunkedIds = _(ids).chunk(chunkSize).value();
+
+        return Future.sequential(
+            chunkedIds.flatMap(idsChunk => {
+                const idsString = idsChunk.join(";");
+
+                return apiToFuture(
+                    this.api.tracker.events.get({
+                        fields: eventFields,
+                        program: AMR_GLASS_PROE_UPLOADS_PROGRAM_ID,
+                        event: idsString,
+                        skipPaging: true,
+                    })
+                ).flatMap((eventsResponse: { instances: D2TrackerEvent[] }) => {
+                    return Future.success(eventsResponse.instances);
+                });
+            })
+        ).flatMap(listOfEvents => Future.success(_(listOfEvents).flatten().value()));
+    }
+
+    private async getEventsWithFilters(params: {
```

Comment:
This function return a promise but always is used using Future.fromPromise.

If you uses future here using apiToFuture is not necessary to transform in callers